doit/dependency.py

```diff
@@ -426,6 +426,26 @@ def get_state(self, dep, current_state):
             'timestamp': TimestampChecker}
 
 
+class DependencyStatus(object):
+    """Result object for Dependency.get_status.
+
+    @ivar status: (str) one of "run", "up-to-date" or "error"
+    """
+
+    def __init__(self):
+        self.status = None
+
+    def _decide(self, status):
```

Comment:
r/`_error_message`/`_error_reason`. And never mind, got rid of underscores here by now.